Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -329,6 +330,49 @@ extension BrowserViewController: URLBarDelegate {
         hideSearchController()
         updateInContentHomePanel(tabManager.selectedTab?.url)
     }
+
+    func urlBar(urlBar: URLBarView, suggestionForText text: String, complete: String? -> Void) {
+        let options = QueryOptions(filter: text, filterType: FilterType.Url, sort: QuerySort.Frecency)
+        profile.history.get(options, complete: { (cursor: Cursor) in
```

Comment:
Yeah, I think you can still keep these decoupled but let the SearchViewController register itself as an autocomplete provider (or delegate, or whatever you have here). Then the BVC just needs to hook the two together when they're created.